src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -731,6 +731,7 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+                    requestUrlPath = PathNormalizer.NormalizeToNFC(requestUrlPath);
```

Comment:
Just mean it currently does:
1, `Unescape` - loop each byte 
2. `GetUtf8String` (`new char[byteLength * 2]`+ `new string()`) + loop each byte -> char
3. `NormalizeToNFC` (`ArrayPool<char>`+ `new string()`) + loop each char

Could do
1. `Unescape` - loop each byte 
2. `NormalizeToNFC` work on the memory blocks (as `.` and `\` are the same bytes in utf8)
3. `GetUtf8String` (`ArrayPool<char>`+ `new string()`)

1 & 2 could be combined, but would be messy.